rust/cubesql/cubesql/src/compile/engine/udf.rs

```diff
@@ -1031,6 +1037,96 @@ pub fn create_current_timestamp_udf() -> ScalarUDF {
     )
 }
 
+macro_rules! parse_timestamp_arr {
+    ($ARR:expr, $ARR_TYPE: ident, $FN_NAME: ident) => {{
+        let arr = $ARR.as_any().downcast_ref::<$ARR_TYPE>();
+        if arr.is_some() {
+            let mut result = Vec::new();
+            let arr = arr.unwrap();
+            for i in 0..arr.len() {
+                result.push(Duration::$FN_NAME(arr.value(i)));
+            }
+
+            Some(result)
+        } else {
+            None
+        }
+    }};
+}
+
+pub fn create_to_char_udf() -> ScalarUDF {
+    let fun: Arc<dyn Fn(&[ColumnarValue]) -> Result<ColumnarValue> + Send + Sync> =
+        Arc::new(move |args: &[ColumnarValue]| {
+            let arr = args[0].clone().into_array(1);
```

Comment:
You should use match to convert ColumnarValue::Scalar(scalar) instead of cloning it.